pennylane_lightning/core/src/simulators/lightning_qubit/gates/cpu_kernels/GateImplementationsLM.hpp

```diff
@@ -363,10 +370,13 @@ class GateImplementationsLM : public PauliGenerator<GateImplementationsLM> {
                 }
             }
         } else {
+            PL_LOOP_PARALLEL(1)
             for (size_t k = 0; k < exp2(num_qubits); k += dim) {
                 for (size_t inner_idx = 0; inner_idx < dim; inner_idx++) {
```

Comment:
Shouldn't this loop have a SIMD pragma?

Reply:
Having combined the collapse and SIMD variants initially, it seemed to give less performance than required, and introduced some additional restrictions. Thus, I tried to be more frugal with the SIMD additions, and simply added the parallel for additions everywhere we loop explicitly over the coefficients.

Reply:
To be clear, I'm thinking having `PL_LOOP_PARALLEL` on the k-loop (which has size `2**N / dim`) and `PL_LOOP_SIMD` on the inner_idx-loop which has size `dim = 8, 16, 32` typically. But if that doesn't help, mark this as resolved.